chemprop/data/scaffold.py

```diff
@@ -12,21 +12,26 @@
 from .data import MoleculeDataset
 
 
-def generate_scaffold(mol: Union[str, Chem.Mol], include_chirality: bool = False) -> str:
+def generate_scaffold(mol: Union[str, Union[Chem.Mol,Tuple[Chem.Mol,Chem.Mol]]], include_chirality: bool = False) -> str:
     """
     Computes the Bemis-Murcko scaffold for a SMILES string.
 
     :param mol: A SMILES or an RDKit molecule.
     :param include_chirality: Whether to include chirality in the computed scaffold..
     :return: The Bemis-Murcko scaffold for the molecule.
     """
-    mol = Chem.MolFromSmiles(mol) if type(mol) == str else mol
+    if type(mol) == tuple:
+        mol = mol[0]
+    elif type(mol) == str and ">" in mol:
+        mol = Chem.MolFromSmiles(mol.split(">")[0])
+    else:
+        mol = Chem.MolFromSmiles(mol) if type(mol) == str else mol
```

Comment:
To my understanding, in chemprop/data/scaffold.py, function `scaffold_split()` we check the number of molecules and raise a ValueError if it is not 1. Then, data.mols is flattened in line 79 (since it was already checked that number_of_molecules==1) (see [here](https://github.com/chemprop/chemprop/commit/2c98bf743cf90893dcc7b9f2b70786a2c7735b2f#diff-daff7c1aa3e0cd29613fa2f4be2ef1944f340b2da3e028a940c0f08dc62e33e7R79) for the corresponding commit, line 79 in chemprop/data/scaffold.py) - unfortunately, this is not very general, and I cannot use the same logic for splitting reactions.

Reply:
Oh, I was not expecting it to be unsupported. Okay, this seems okay then. 

I feel a little odd about the explicit check for ">" in a string in the middle of something else. It seems like maybe there's an opportunity to have a utility function elsewhere that you can call to do that check (is_reaction_smiles() or something) and then have it available if you ever need that check elsewhere. This is fine though if that sounds unnecessary.